tests/torch/ptq/test_weights_compression.py

```diff
@@ -327,3 +354,72 @@ def test_pack_int4():
     assert packed_w.numel() * 2 == w_int8.numel()
     unpacked_w = unpack_int4(packed_w).reshape(w_int8.shape)
     assert torch.all(unpacked_w == w_int8)
+
+
+class TestPTTemplateWeightCompression(TemplateWeightCompression):
+    @staticmethod
+    def get_matmul_model() -> torch.nn.Module:
+        return MatMulModel(255 * torch.eye(3, dtype=torch.float32))
+
+    @staticmethod
+    def get_sequential_matmul_model() -> torch.nn.Module:
+        return SequentialMatmulModel()
+
+    @staticmethod
+    def to_tensor(t) -> torch.Tensor:
+        return torch.tensor(t)
+
+    @staticmethod
+    def cast_to(x: torch.Tensor, dtype: TensorDataType) -> torch.Tensor:
+        return cast_to(x, dtype)
+
+    @staticmethod
+    def check_weights(model: torch.nn.Module, ref_ids: List[int]) -> None:
+        for i, op in enumerate(model.layers):
+            if i in ref_ids:
+                assert torch.numel(op.weight) == 8  # workaround to detect uint4 weights
+            else:
+                assert torch.numel(op.weight) == 16
```

Comment:
This function looks too hardcoded for a particular model. 
What about converting torch model to OV here and explicitly check for ov.Type.i4?

Reply:
updated, please check